icefoganalytics/travel-authorization
# Update Per-Diem To Compute Based Off Final Destination

Fixes https://github.com/icefoganalytics/travel-authorization/issues/127

Relates to:
- https://governmentofyukon.slack.com/archives/C04ETKKJS1X/p1701707549249019

# Context

Current per-diems calculations are based of hard to explain factors, and don't really make sense. This PR will update the per-diems calculation so they are based of the final destination

e.g.
If I travel from Whitehorse to Vancouver, to Los Angeles, would the per-diems be **for Los Angeles only**.

This will require a re-work of
- `api/src/services/estimates/bulk-generate/determine-location-from-date.ts`

# Implementation

Use travel segment information to determine trip type, and use trip type to determine which value to calculate the final destination from.
Fix bug in Multi-Destination trip data entry form pre-filling. See https://github.com/icefoganalytics/travel-authorization/pull/152/commits/eac3d4b6166564d68a891547f08fcc51e888965d

## Concerns

We don't currently have international travel support.
The `locations` table doesn't even have a field for country.

# Screenshots

I love tests
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/5385c9e4-b389-48d1-b0f2-b3984de2ceba)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Go to the "My Travel Requests" page via the top drop down nav.
5. Create a travel request via the "+ Travel Authorization" button.
6. Create a round trip from Whitehorse to Vancouver from the 5th to the 7th with the final departure time of 15:00 hours.
7. Scroll down, then generate the estimates and then view them.
8. Check that the Meals & Incidentals values look correct.
9. Repeat these steps for a One Way trip to Vancouver.
10. Finally repeat these steps for a Multi-Destination trip from Whitehorse to Vancouver to Grande Prairie (AB).
